Register REST API and forward REST requests to associated extension (#101)
opensearch-project/opensearch-sdk-java: DESIGN.md

Immediately following initialization, each extension will establish a connection to OpenSearch on its own transport service, and send its REST API (a list of methods and URIs to which it will respond).  These will be registered with the RestController.

When OpenSearch receives a registered method and URI, it will send the request to the Extension. The extension will appropriately handle the request, using the API to determine which Action to execute.

### OpenSearch SDK for Java

Currently, plugins rely on extension points to communicate with OpenSearch. These are represented as Actions. To turn plugins into extensions, the Extension must assemble a list of all methods and URIs to communicate to OpenSearch, where they will be registered; upon receiving a matching request from a user these will be forwarded back to the Extension and the Extension will further need to handle these registered methods and URIs with an appropriate Action.

### Extension Walk Through

1. Extensions are started up and must be running before OpenSearch is started.  (In the future, there will be a facility to refresh the extension list during operation and handle network communication interruptions.)

> 2. OpenSearch is started. During its bootstrap, the `ExtensionsOrchestrator` is initialized, reading extension definitions from `extensions.yml`.

Reviewer: ```suggestion
2. OpenSearch is started. During its bootstrap, the `ExtensionsOrchestrator` is initialized, reading list of extension present in `extensions.yml`.
```
